src/lambdas/RedriveSNSDQLEvents/helper.ts

```diff
@@ -0,0 +1,76 @@
+import { SQSBatchItemFailure, SQSEvent } from 'aws-lambda'
+import { AuditEvent } from '../../types/auditEvent'
+import { tryParseJSON } from '../../utils/helpers/tryParseJson'
+
+export type ProcessingResult = {
+  sqsMessageId: string
+  failed: boolean
+  statusReason: string
+  auditEvent?: AuditEvent
+}
+
+export const SQSBatchItemFailureFromProcessingResultArray = (
+  processingResultArray: ProcessingResult[]
+): SQSBatchItemFailure[] => {
+  return processingResultArray.map((element) => {
+    return { itemIdentifier: element.sqsMessageId }
+  })
+}
+
+export const generateEventIdLogMessageFromProcessingResult = (
+  processingResultArrays: ProcessingResult[][]
+) => {
+  const logMessage: { [key: string]: string[] } = {}
+  processingResultArrays.forEach((singleProcessingResultArray) => {
+    singleProcessingResultArray.forEach((processsingResult) => {
+      if (!Object.keys(logMessage).includes(processsingResult.statusReason)) {
+        logMessage[`${processsingResult.statusReason}`] = []
+      }
+      logMessage[`${processsingResult.statusReason}`].push(
+        processsingResult.auditEvent?.event_id as string
+      )
+    })
+  })
+
+  return logMessage
+}
+
+export const parseSQSEvent = (event: SQSEvent) => {
```

Comment:
I know this is probably a bit pedantic but don't we normally try to declare return type explicitly?

Reply:
sorted